dali/pipeline/operators/geometric/bb_flip_test.cc

```diff
@@ -33,213 +38,157 @@ constexpr int kBbStructSize = 4;
-        }
+const TestSample rois_ltrb[] = {
+        {{.2,  .2,  .6,  .5},  {.4,  .2,  .8,  .5},  {.2,  .5,  .6,  .8}},
+        {{.0,  .0,  .5,  .5},  {.5,  .0,  1.,  .5},  {.0,  .5,  .5,  1.}},
+        {{.3,  .2,  .4,  .3},  {.6,  .2,  .7,  .3},  {.3,  .7,  .4,  .8}},
+        {{.0,  .0,  .2,  .3},  {.8,  .0,  1.,  .3},  {.0,  .7,  .2,  1.}},
+        {{.0,  .0,  .1,  .1},  {.9,  .0,  1.,  .1},  {.0,  .9,  .1,  1.}},
+        {{.5,  .5,  .6,  .6},  {.4,  .5,  .5,  .6},  {.5,  .4,  .6,  .5}},
+        {{.0,  .6,  .7,  .9},  {.3,  .6,  1.,  .9},  {.0,  .1,  .7,  .4}},
+        {{.6,  .2,  .9,  .5},  {.1,  .2,  .4,  .5},  {.6,  .5,  .9,  .8}},
+        {{.4,  .3,  .9,  .8},  {.1,  .3,  .6,  .8},  {.4,  .2,  .9,  .7}},
+        {{.25, .25, .75, .75}, {.25, .25, .75, .75}, {.25, .25, .75, .75}},
 };
 
 
-/**
- * Injects either input values of test std::arrays (i.e. left-hand
- * Rois) or anticipated output values, which is the reference data.
- * @tparam DataIdx Index of tuple contained in TestData. Specifies
- *                 which values to inject
- */
-template<typename DataType, int DataIdx>
-void InjectTestData(const TestData &test_data, DataType *destination) {
-  for (const auto &it : test_data) {
-    auto _it = std::get<DataIdx>(it).data();
-    std::memcpy(destination, _it, kBbStructSize * sizeof(DataType));
-    destination += kBbStructSize;
+template<size_t N>
+const TestSample &FindSample(const TestSample (&dataset)[N], const Roi &roi) {
+  for (auto &sample : dataset) {
+    if (sample[0] == roi) {
+      return sample;
+    }
   }
+  DALI_FAIL("TestSample for provided `roi` has not been found");
 }
 
-}  // namespace
 
-template<typename ImageType>
-class BbFlipTest : public DALISingleOpTest<ImageType> {
- protected:
-  std::vector<TensorList<CPUBackend> *>
-  Reference(const std::vector<TensorList<CPUBackend> *> &inputs, DeviceWorkspace *ws) override {
-    using DataType = float;
-    auto batch = new TensorList<CPUBackend>();
-    batch->Resize(new_batch_size_);
-    auto *batch_data = batch->template mutable_data<DataType>();
-
-    DALI_ENFORCE(!(flip_type_horizontal_ && flip_type_vertical_), "No test data for combined case");
-
-    if (flip_type_vertical_) {
-      InjectTestData<DataType, 2>(*test_data_, batch_data);
-    } else if (flip_type_horizontal_) {
-      InjectTestData<DataType, 1>(*test_data_, batch_data);
-    } else {
-      InjectTestData<DataType, 0>(*test_data_, batch_data);
+/**
+ * Create TensorList from TestSample data
+ *
+ * @tparam Idx index of data @see TestSample
+ * @tparam Backend
+ * @tparam N size of TestSample data
+ * @param sample
+ * @return
+ */
+template<size_t Idx, typename Backend, size_t N>
+std::unique_ptr<TensorList<Backend>> ToTensorLists(const TestSample (&sample)[N]) {
+  std::unique_ptr<TensorList<Backend>> tl(new TensorList<Backend>());
+  tl->Resize({N, {kBbStructSize}});
+  auto ptr = tl->template mutable_data<float>();  // TensorList type initialization
```

Comment:
This comment doesn't clarify anything. If the reader is familiar enough with the code to know that `TensorList` needs to have "type initialization", they already know that `mutable_data` does set the type.

Reply:
removed comment